### PR TITLE
[codex] Add Garmin lap details route map

### PR DESCRIPTION
**Pull request by stuartshay**
## Summary

Refs #274

This expands the Garmin Activity Laps tab so selected laps show detailed recorded stats and a route map for the selected segment.

Changes include:
- Inline selected-lap details panel below the laps table.
- Recorded lap stats for timer/elapsed/moving time, distance, paved/unpaved %, speed, heart rate, ascent/descent, calories, and start/end time.
- Route segment rendering using existing chart track points and `ActivityRouteMap`.
- Robust lap route segmentation that falls back from lap `end_time` to `start_time + duration_seconds`, then to cumulative lap distance when timestamps cannot resolve enough GPS points.
- Unit/page/e2e coverage for lap selection, details, route segmentation, and empty route state.

## Root Cause / Context

The initial lap route approach used lap `start_time` and `end_time` directly. Some Garmin lap rows return invalid `end_time` values, including values equal to the activity start, so timestamp-only filtering can produce zero or one route point even when the activity has full GPS data.

## Validation

- `npm run test:run -- src/components/garmin/ActivityLapsTable.test.tsx src/pages/GarminDetailPage.test.tsx`
- `npm run type-check`
- `npm run lint:all`
- `npm run build`

Build passed with existing Vite chunk-size/deprecation warnings only.

### CODE REVIEW

**Comment by stuartshay** [comment]
Implementation summary:
- Added inline selected-lap details panel under the Laps table.
- Added selected lap route rendering via existing chart track points and ActivityRouteMap.
- Added robust segment fallback for invalid Garmin lap end_time values: duration first, cumulative distance second.
- Added unit/page/e2e coverage for lap details and route/empty states.

Validation performed:
- npm run test:run -- src/components/garmin/ActivityLapsTable.test.tsx src/pages/GarminDetailPage.test.tsx
- npm run type-check
- npm run lint:all
- npm run build

Notes: build passed with existing Vite chunk-size/deprecation warnings only.

**Comment by stuartshay** [comment]
Addressed Copilot review comments in 0efc891:
- Displayed lap end time now uses the same effective fallback window as route segmentation.
- One resolved lap route point now renders the map instead of the empty route state.
- Distance fallback now aborts when any prior lap distance is missing/invalid, avoiding wrong cumulative offsets.
- Renamed the misleading empty-segment test and added coverage for the new edge cases.

Validation rerun:
- npm run test:run -- src/components/garmin/ActivityLapsTable.test.tsx src/pages/GarminDetailPage.test.tsx
- npm run type-check
- npm run lint:all
- npm run build

File: src/components/garmin/ActivityLapsTable.tsx [hunk 1]
```diff
@@ -53,7 +97,127 @@ function surfacePercent(
...
+    <div className="space-y-4" data-testid="activity-lap-details-panel">
+      <Card>
+        <CardContent className="space-y-5 p-4">
+          <div className="flex flex-wrap items-start justify-between gap-3">
+            <div>
+              <h3 className="text-lg font-semibold">Lap {lap.lap_index}</h3>
+              <p className="text-sm text-muted-foreground">
+                {formatTimestamp(lap.start_time)} –{' '}
+                {formatTimestamp(lap.end_time)}
+              </p>
```
[the remark this answers is not in the document]
Comment (stuartshay) [reply src/components/garmin/ActivityLapsTable.tsx:138]: Addressed in 0efc891. The details panel now uses `getLapTimeWindow`, the same effective fallback window used by route segmentation, so invalid raw `lap.end_time` values are no longer displayed as the selected lap end time.

File: src/components/garmin/ActivityLapsTable.test.tsx [hunk 2]
```diff
@@ -15,11 +42,164 @@ function lap(overrides: Partial<ActivityLap> = {}): ActivityLap {
...
+      },
+    ])
+
+    expect(points.map((point) => point.timestamp)).toEqual([
+      '2026-03-14T09:05:00Z',
+      '2026-03-14T09:10:00Z',
+    ])
+  })
+
+  it('returns no lap segment points when lap timestamps are missing or invalid', () => {
```
[the remark this answers is not in the document]
Comment (stuartshay) [reply src/components/garmin/ActivityLapsTable.test.tsx]: Addressed in 0efc891. Renamed the test to clarify it covers the empty case when timestamp and distance fallback data are unavailable, and added separate coverage for distance fallback behavior.

File: src/components/garmin/ActivityLapsTable.tsx [hunk 3]
```diff
@@ -53,7 +98,132 @@ function surfacePercent(
...
+  )
+}
+
+function toRoutePoints(points: ActivityChartTrackPoint[]): LapRoutePoint[] {
+  return points.map((point) => ({
+    latitude: point.latitude as number,
+    longitude: point.longitude as number,
+    speed_kmh: point.speed_kmh,
+  }))
+}
```
[the remark this answers is not in the document]
Comment (stuartshay) [reply src/components/garmin/ActivityLapsTable.tsx]: Addressed in 9c3a193. Moved lap route conversion into a dedicated helper that defensively filters/type-narrows to finite latitude/longitude before mapping to ActivityRouteMap points, and added focused coverage for invalid/null coordinates. Validation passed: targeted Vitest suite, type-check, lint:all, and build.
